src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
-}
+  const clone = [];
+  let currentState = { ...state };
+
+  for (const object of actions) {
+    const { type, extraData = {}, keysToRemove = [] } = object;
+
+    if (type === 'addProperties') {
+      currentState = { ...currentState, ...extraData };
+    }
+
+    if (type === 'removeProperties') {
+      currentState = { ...currentState };
 
+      for (const key of keysToRemove) {
+        delete currentState[key];
+      }
+    }
+
+    if (type === 'clear') {
+      currentState = {};
+    }
+    clone.push(currentState);
```

Comment:
[DONT REPEAT YOURSELF] - Instead of pushing the 'currentState' into the 'clone' array within each condition, do it just once after the conditions have been checked. This will avoid repetition and make the code more maintainable.